force-app/main/default/classes/ContactRollupsSchedulable_TEST.cls

```diff
@@ -20,43 +26,210 @@ private class ContactRollupsSchedulable_TEST {
+        System.assertEquals(1, contacts[0].NumPresentServiceDeliveries__c);
+        System.assertEquals(2, contacts[0].NumAbsentServiceDeliveries__c);
+        System.assertEquals(2, contacts[0].ConsecutiveAbsences__c);
+        System.assertEquals(System.today(), contacts[0].LastServiceDate__c);
+        System.assertEquals(null, contacts[1].NumPresentServiceDeliveries__c);
+        System.assertEquals(null, contacts[1].NumAbsentServiceDeliveries__c);
+        System.assertEquals(null, contacts[1].ConsecutiveAbsences__c);
+    }
+
+    @IsTest
+    private static void shouldDecrementCountOnDelete() {
+        testDataSetup();
+
+        Test.startTest();
+        delete serviceDeliveries[0];
+
+        ContactRollupsSchedulable contactRollups = new ContactRollupsSchedulable();
+        Database.executeBatch(contactRollups);
 
         Test.stopTest();
+        requery();
+
+        System.assertEquals(2, contacts[0].NumPresentServiceDeliveries__c);
+        System.assertEquals(0, contacts[0].NumAbsentServiceDeliveries__c);
+        System.assertEquals(0, contacts[0].ConsecutiveAbsences__c);
+        System.assertEquals(System.today(), contacts[0].LastServiceDate__c);
+    }
 
-        attendanceServiceStub.assertCalledWith(
-            'doRollups',
-            new List<Type>{ SObjectType.class, List<SObject>.class },
-            new List<Object>{ Contact.SObjectType, contactsWithServiceDeliveries }
+    @IsTest
+    private static void shouldScheduleContactRollupsBatch() {
+        Test.startTest();
+
+        //parse to cron expression
+        String scheduledTime = '0 0 12 * * ?';
+
+        ContactRollupsSchedulable contactRollups = new ContactRollupsSchedulable();
+        String jobId = System.schedule(
+            'Job Started At ' + scheduledTime,
+            scheduledTime,
+            contactRollups
         );
+
+        CronTrigger cronTrigger = [
+            SELECT Id, CronExpression, TimesTriggered
+            FROM CronTrigger
+            WHERE Id = :jobId
+        ];
+
+        //Assert the job has not run yet
+        System.assertEquals(0, cronTrigger.TimesTriggered);
+
+        Test.stopTest();
+    }
+
+    private static void testDataSetup() {
+        requery();
+
+        System.assertEquals(null, contacts[0].NumPresentServiceDeliveries__c);
+        System.assertEquals(null, contacts[0].NumAbsentServiceDeliveries__c);
+        System.assertEquals(null, contacts[0].ConsecutiveAbsences__c);
+        System.assertEquals(null, contacts[0].LastServiceDate__c);
+
+        Contact newContact = new Contact();
+        newContact.LastName = 'Test Batch Contact';
+        newContact.AccountId = accountRecordId;
+        newContact.NumAbsentServiceDeliveries__c = 0;
+        newContact.NumPresentServiceDeliveries__c = 1;
+        newContact.ConsecutiveAbsences__c = 0;
+        newContact.LastServiceDate__c = System.today();
+        insert newContact;
+
+        ProgramEngagement__c engagement = new ProgramEngagement__c();
```

Comment:
Do we need to be concerned with creating additional engagements in the contact test? If we create these for every test can it just be part of test setup?

Reply:
Just thinking out loud no biggy if we want to leave as is.